### DISCUSSION TITLE
Jursin 的提议集

### DISCUSSION THREAD

**Discussion by DryIce-cc** [ClassIsland/ClassIsland#334] [常规]
对于以下功能优化，issue或discuss中基本没看到重复的，看看这些是否合理，有没有提issue的必要

- [ ] 1.组件单独圆角

- [ ] 2.已上课程淡化

- [ ] 3.托盘图标增加重启按钮

- [ ] 4.档案编辑界面增加撤回与恢复按钮

- [ ] 5.调整时间表、课程表顺序
> ( : ) 回应：这个不好弄（）用Dictionary 就会有这个问题，得换其它有序的字典类型，比如OrderedDictionary 
但ordereddictionary不是很好用，好像还没有泛型版本的 --HelloWRC (干冰DryIce 转发)

- [ ] 6.适当调大软件默认缩放大小
> (－) 反对：缩放大小在不同设备上大小差距较大，已在欢迎向导中提供了滑块，修改默认值无意义。--干冰DryIce

- [ ] 7.提醒时自动将音量设为100
> (～) 补充：考虑到系统音量可能会有别的应用正在使用，建议 ClassIsland 另建虚拟音频设备。--干冰DryIce

- [ ] 8.自定义提醒横幅停留时间
> (+) 支持：建议「提醒」参考类似组件的编排方式，支持用户自定义。--干冰DryIce

- [ ] 9.支持跳转到指定时刻触发指定提醒以测试

**Comment by liuziheng20091106** [comment]
合理。
缩放建议弄两个，一个是默认模式，一个是提醒时的缩放，这样提醒时稍微放大，方便后排同学看

**Reply by 6JinQuan** [reply]
再加一个：类似于隔壁ecs，上课时仅显示倒计时并且缩小界面（现在是只能设置规则使得上课时只显示当科课程+倒计时，但是首先界面太大比较遮挡视线，其次上课期间没必要显示课程，只显示倒计时即可，这样方便同学老师把握上课速度和进度）同时倒计时上有个负号“-”感觉有点不符合常规使用观感，建议改进。

**Comment by Jursin** [comment]
关于1.的描述补充
类似 ElectronClassSchedule，将一个或多个合并组件分别在收尾两端绘制圆角（可能描述不清，就是下面这样）
![preview.jpg](https://github.com/user-attachments/assets/c1d66b79-56ae-45eb-a2ed-87490c67095d)
补充建议
- [ ] 课程允许添加角标
- [ ] 组件支持双行显示
- [ ] 字体设为圆体后字重无法进一步调大

**Reply by DryIce-cc** [reply]
这个建议可以，我也是这么想的，在「分割线」组件中加入「分岛」选项。

> 课程允许添加角标

这个我也想要，不过感觉 ecs 那边实际做出来，看着有点不协调。
ci 的字本来就小小小小小，再角标可能会看不清，虽然时间点附加信息也看不清（

> 字体设为圆体后字重无法进一步调大

应该是这个字体本身不支持更大的字重。

**Reply by Jursin** [reply]
同样的字体「幼圆」在ECS上显示的清楚些，但在CI上感觉有点细🤔

**Comment by wang-lu-yuan** [comment]
对于提醒时自动设置音量为100%这个建议（的补充说明），我觉得可以通过音量混合器（win7音量设置中的“合成器”）实现
提醒时，先将系统音量设置为100，然后在音量混合器里将其他应用的音量设为原来的音量，播放完提醒之后再将系统音量恢复，这个思路不知如何？

**Reply by liuziheng20091106** [reply]
啊，感觉有点太麻烦了。
不如开音频直通（开玩笑的）
虚拟音频设备的思路不错，不过依然麻烦。
这东西似乎没有简单的解决方案

**Reply by liuziheng20091106** [reply]
Windows是会记忆合成器里的音频设置的，有时候如果调了合成器设置你这么操作会覆盖原有设置，在某些情况下甚至不如全局调100再调回去

**Reply by wang-lu-yuan** [reply]
> 啊，感觉有点太麻烦了。
> 不如开音频直通（开玩笑的）
> 虚拟音频设备的思路不错，不过依然麻烦。
> 这东西似乎没有简单的解决方案

照这样的话，这肯定就没有简单的解决方案了。不如直接用Windows自己的提示（但改成classisland的提示音）

**Reply by wang-lu-yuan** [reply]
> Windows是会记忆合成器里的音频设置的，有时候如果调了合成器设置你这么操作会覆盖原有设置，在某些情况下甚至不如全局调100再调回去

现用现调，用完调回去，有问题再调:)

**Comment by Jursin** [comment]
> 对于以下功能优化，issue或discuss中基本没看到重复的，看看这些是否合理，有没有提issue的必要
> 
> - [ ] 1.组件单独圆角
> 
> - [ ] 2.已上课程淡化
> 
> - [ ] 3.托盘图标增加重启按钮
> 
> - [ ] 4.档案编辑界面增加撤回与恢复按钮
> 
> - [ ] 5.调整时间表、课程表顺序
> > ( : ) 回应：这个不好弄（）用Dictionary 就会有这个问题，得换其它有序的字典类型，比如OrderedDictionary 
> 但ordereddictionary不是很好用，好像还没有泛型版本的 --HelloWRC (干冰DryIce 转发)
> 
> - [ ] 6.适当调大软件默认缩放大小
> > (－) 反对：缩放大小在不同设备上大小差距较大，已在欢迎向导中提供了滑块，修改默认值无意义。--干冰DryIce
> 
> - [ ] 7.提醒时自动将音量设为100
> > (～) 补充：考虑到系统音量可能会有别的应用正在使用，建议 ClassIsland 另建虚拟音频设备。--干冰DryIce
> 
> - [ ] 8.自定义提醒横幅停留时间
> > (+) 支持：建议「提醒」参考类似组件的编排方式，支持用户自定义。--干冰DryIce
> 
> - [ ] 9.支持跳转到指定时刻触发指定提醒以测试

- 主界面支持云母等更多磨砂玻璃模糊特效
- 字体大小添加滑动条控制

**Comment by 6JinQuan** [comment]
关于第2点个人补充如下：在课表信息较多、课程较多的情况下，课程信息在屏幕上难以完整显示（兼并字号大，后排能看清的情况），我建议已上课程淡化合并，和前面其它已结束课程可以合并为一个集，可设置特定时间显示，或者课间定期显示（如显示1分钟后缩成一个集），单击这个集的按钮（如：“ ||| ”）或者悬停时可以展开，其他时间按照规则进行。




关于第7点：同时可以换个思路，在下课时刻向老师发出一个1～3秒的居中弹窗，老师可以自行关闭，但同时也能起到提醒作用。



同时不同功能改变应在设置中提供一定的开关选项，供不同使用用户需求自行选择使用。


其他项同上无异议。



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年10月19日(周六) 晚上10:17
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [ClassIsland/ClassIsland] Jursin 的提议集 (Discussion #334)




  
对于以下功能优化，issue或discuss中基本没看到重复的，看看这些是否合理，有没有提issue的必要
  
 
 1.组件单独圆角
 
 
 
 2.已上课程淡化
 
 
 
 3.托盘图标增加重启按钮
 
 
 
 4.档案编辑界面增加撤回与恢复按钮
 
 
 
 5.调整时间表、课程表顺序
 
   
( : ) 回应：这个不好弄（）用Dictionary 就会有这个问题，得换其它有序的字典类型，比如OrderedDictionary
 但ordereddictionary不是很好用，好像还没有泛型版本的 --HelloWRC (干冰DryIce 转发)
   
 6.适当调大软件默认缩放大小
   
(－) 反对：缩放大小在不同设备上大小差距较大，已在欢迎向导中提供了滑块，修改默认值无意义。--干冰DryIce
   
 7.提醒时自动将音量设为100
   
(～) 补充：考虑到系统音量可能会有别的应用正在使用，建议 ClassIsland 另建虚拟音频设备。--干冰DryIce
   
 8.自定义提醒横幅停留时间
   
(+) 支持：建议「提醒」参考类似组件的编排方式，支持用户自定义。--干冰DryIce
   
 9.支持跳转到指定时刻触发指定提醒以测试
    
主界面支持云母等更多磨砂玻璃模糊特效
 
字体大小添加滑动条控制
